pallets/identity/src/lib.rs

```diff
@@ -2226,42 +2226,65 @@ impl<T: Trait> CheckAccountCallPermissions<T::AccountId> for Module<T> {
         pallet_name: &PalletName,
         function_name: &DispatchableName,
     ) -> Option<AccountCallPermissionsData<T::AccountId>> {
+        // `who` is the original origin of the call (including forwarded calls from an identity acting as a key)
+        // context::current_identity is the target did against which the calling key must have permissions
+        // if `who`'s identity does not match context::current_identity we have a forwarded call, in this case the
+        // key for which we check permissions is `who`'s identity rather than `who`
         if <KeyToIdentityIds<T>>::contains_key(who) {
```

Comment:
This is a big `if` block. Maybe make the `else` branch return first to remove unnecessary indentation?
```
if !<KeyToIdentityIds<T>>::contains_key(who) {
    return None;
}
...
```